src/createReverseGeocodingSource.js

```diff
@@ -0,0 +1,119 @@
+import formatHit from './formatHit';
+import configure from './configure';
+import defaultTemplates from './defaultTemplates';
+
+const filterApplicableParams = params => {
+  const { hitsPerPage, aroundLatLng, getRankingInfo, language } = params;
+
+  const filtered = {};
+
+  if (typeof hitsPerPage === 'number') {
+    filtered.hitsPerPage = hitsPerPage;
+  }
+
+  if (typeof language === 'string') {
+    filtered.language = language;
+  }
+
+  if (typeof getRankingInfo === 'boolean') {
+    filtered.getRankingInfo = getRankingInfo;
+  }
+  if (typeof aroundLatLng === 'string') {
+    filtered.aroundLatLng = aroundLatLng;
+  }
+
+  return filtered;
+};
+
+const createReverseGeocodingSource = ({
+  apiKey,
+  appId,
+  hitsPerPage,
+  aroundLatLng,
+  getRankingInfo,
+  formatInputValue = defaultTemplates.value,
+  language = navigator.language.split('-')[0],
+  onHits = () => {},
+  onError = e => {
+    throw e;
+  },
+  onRateLimitReached,
+}) => {
+  const configuration = configure({
+    apiKey,
+    appId,
+    hitsPerPage,
+    aroundLatLng,
+    getRankingInfo,
+    language,
+    formatInputValue,
+    onHits,
+    onError,
+    onRateLimitReached,
+  });
+
+  let params = filterApplicableParams(configuration.params);
+  let controls = configuration.controls;
+
+  const baseUrl = `https://places-dsn.algolia.net/1/places/reverse?x-algolia-application-id=${appId}&x-algolia-api-key=${apiKey}`;
```

Comment:
We probably can do better here. The retry strategy won't work here if the DNS is down (which happens). Why don't we leverage the client here?

Reply:
I don't think the client has that route (`/places/reverse`) implemented yet, which is why I couldn't use it. If it is available, I'll use it!

Last I checked, the client only had exposed `.getObject` and `.search` method:
![image](https://user-images.githubusercontent.com/5136989/48856238-c0b1e380-ed6a-11e8-939c-f210a1571d05.png)

Reply:
`.reverse` method implemented has been added in PR: https://github.com/algolia/algoliasearch-client-javascript/pull/732
Waiting for approval.